### PR TITLE
Handle click on follow notification

### PR DESCRIPTION
**Pull request by zanderz**
This is still waiting on a server tweak to send the username instead of user ID, so it still hasn't been fully exercised. The server still only sends the notification for 3 hardcoded user ids anyway. Does this seem like the right way to show the user?

@keybase/react-hackers

### CODE REVIEW

File: shared/actions/push/index.js [hunk 1]
```diff
@@ -60,15 +61,27 @@ function* pushNotificationSaga(notification: Constants.PushNotification): SagaGe
 
     // Check for conversation ID so we know where to navigate to
     const convID = payload.data ? payload.data.convID : payload.convID
-    if (!convID) {
-      console.error('Push notification payload missing conversation ID')
+    const type = payload.type
+    if (convID) {
```
Comment (chrisnojima) [inline shared/actions/push/index.js]: now we have type that should be checked first

Comment (chrisnojima) [inline shared/actions/push/index.js]: like this should be like type='chat' or something. what is it @mmaxim

**Comment by chrisnojima** [comment]
we shouldn't merge this until core makes the change to send the username

**Comment by mmaxim** [comment]
That username change is live, I put it live yesterday.

File: shared/constants/push.js [hunk 2]
```diff
@@ -13,10 +13,13 @@ export type PushNotification = {
   payload?: {
     userInteraction: boolean,
     convID?: string, // Android variant
+    type?: string,
```
Comment (chrisnojima) [inline shared/constants/push.js]: we can have a better type here:

```


export type PushNotification = {
    payload?: {
        userInteraction: boolean,
        // Android variant
        convID?: string, 
        type: 'chat.newmessage',
        data?: {
            // iOS variant
            convID?: string,
            type: 'chat.newmessage',
        },
    },
} | {
    payload?: {
        userInteraction: boolean,
        type: 'follow',
        username?: string,
    },
}

```

something like that

Comment (mmaxim) [inline shared/constants/push.js]: Yeah we could also put this in AVDL, probably should have done that from the beginning. It's probably cool to just put the type in manually, and I can ticket moving the payload types into AVDL so we can just get it for free.

**Comment by chrisnojima** [comment]
@zanderz lets just merge this

**Pull request merged**